### PR TITLE
Add BAS GATT reporting

### LINKED ISSUE #47: Feature: Proper battery reporting via BLE battery service
ZMK already exposes the standard BLE battery service, but doesn't have any code to actually send values to the connected host. Needs a generic implementation that can be plugged for different keyboards as needed.

### PR DESCRIPTION
**Pull request by Nicell**
This adds BAS GATT reporting using any device binding with the label `BATTERY`. Closes #47.

Notes:

- `bt_gatt_bas_set_battery_level` has been renamed in Zephyr 2.4.0 -> #223 
- `zmk_battery_update` should eventually raise a battery event instead of directly posting to the BAS GATT
- If the board doesn't have a battery reporting device, init will fail and none of the timers will run. Should this be behind a Kconfig value instead?
- I'm thinking that the update function should only update if the battery value has changed since last time. This would for example save from an unnecessary ePaper display refresh in the future.

### CODE REVIEW

File: app/src/battery.c [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+#include <drivers/sensor.h>
+#include <bluetooth/services/bas.h>
+
+#include <logging/log.h>
+
+LOG_MODULE_DECLARE(zmk, CONFIG_ZMK_LOG_LEVEL);
+
+struct device *battery;
+
+static int zmk_battery_update(struct device *battery) {
```
Comment (Nicell) [inline app/src/battery.c:19]: Should I make this function available to the rest of the application? I guess that's the reason I have the `battery` device passed in to here.

Comment (petejohanson) [reply app/src/battery.c:19]: I don't see why we would need to, at least not to begin with. This should be able to be pretty self contained.

File: app/src/battery.c [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+static int zmk_battery_init(struct device *_arg) {
+    battery = device_get_binding("BATTERY");
+
+    if (battery) {
+        LOG_DBG("Found battery reporting device.");
+    } else {
+        LOG_DBG("No battery device labelled BATTERY found.");
+        return -ENODEV;
+    }
```
Comment (petejohanson) [inline app/src/battery.c:63]: We typically don't bother logging or branching for the success case. Usually we'd just do:

```suggestion
    if (battery == NULL) {
        LOG_DBG("No battery device labelled BATTERY found.");
        return -ENODEV;
    }
```

**Review by petejohanson — COMMENTED** [review]
Two super minor comments.

**Review by petejohanson — APPROVED** [review]
Thanks!

**Comment by Katona** [comment]
Hi! Would it be possible to report the battery level of both half in case of split keyboards? Either through the "master" (maybe with pressing some combo) or directly to the host via Bluetooth (if it's possible at all)?

**Pull request merged**